### PR TITLE
Describe one variable re-implementation

### PR DESCRIPTION
**Pull request by stevekogo**

### CODE REVIEW

**Comment by stevekogo** [comment]
@africanmathsinitiative/instat  The checkboxes in the subdialogue feeds the sub routine SummariesParameters() to create a list of summaries using the tags. Setting this to wok properly has been quite a challenge and I welcome suggestions on this.

**Comment by dannyparsons** [comment]
@africanmathsinitiative/developers Can someone review this?

File: instat/sdgSummaries.Designer.vb [hunk 1]
```diff
@@ -256,30 +215,27 @@ Partial Class sdgSummaries
...
+    Friend WithEvents ucrChkMinimum As ucrCheck
+    Friend WithEvents ucrChkMaximum As ucrCheck
+    Friend WithEvents ucrChkSum As ucrCheck
+    Friend WithEvents ucrChkMean As ucrCheck
+    Friend WithEvents ucrChkStdDev As ucrCheck
+    Friend WithEvents ucrChkRange As ucrCheck
+    Friend WithEvents ucrChkVariance As ucrCheck
+    Friend WithEvents ucrChkMedian As ucrCheck
+    Friend WithEvents ucrChkQuartiles As ucrCheck
 End Class
```
Comment (lilyclements) [inline instat/sdgSummaries.Designer.vb]: In general this looks good. There are a few issues with the check boxes not being aligned the same and also not being spaced apart equally. Finally, the checkbox cuts off some of the group box - either the group boxes should be made bigger or the checkboxes smaller (if possible!)

File: instat/dlgDescribeOneVariable.vb [hunk 2]
```diff
@@ -13,11 +13,14 @@
...
 Imports instat.Translations
+
 Public Class dlgDescribeOneVariable
     Public bFirstLoad As Boolean = True
-    Public clsRBaseFunction, clsRCustomFunction As New RFunction
+    Private clsDefaultFunction As New RFunction
+    Public clsRCustomFunction As New RFunction
+    Private bResetSubdialog As Boolean = False
     Private Sub dlgDescriptiveStatistics_Load(sender As Object, e As EventArgs) Handles MyBase.Load
         If bFirstLoad Then
```
Comment (lilyclements) [inline instat/dlgDescribeOneVariable.vb]: The _load needs to be altered to something along the lines of

        autoTranslate(Me)
        If bFirstLoad Then
            InitialiseDialog()
            bFirstLoad = False
        End If
        If bReset Then
            SetDefaults()
        End If
        SetRCodeForControls(bReset)
        bReset = False
        ReopenDialog()
        TestOkEnabled()
    End Sub

File: instat/dlgDescribeOneVariable.vb [hunk 3]
```diff
@@ -13,11 +13,14 @@
 '
 ' You should have received a copy of the GNU General Public License k
 ' along with this program.  If not, see <http://www.gnu.org/licenses/>.
-
+Imports instat
 Imports instat.Translations
+
 Public Class dlgDescribeOneVariable
     Public bFirstLoad As Boolean = True
```
Comment (lilyclements) [inline instat/dlgDescribeOneVariable.vb]: We want to add in Private bReset here

File: instat/dlgDescribeOneVariable.vb [hunk 4]
```diff
@@ -29,120 +32,97 @@ Public Class dlgDescribeOneVariable
...
-    Private Sub cmdSummaries_click(sender As Object, e As EventArgs) Handles cmdSummaries.Click
-        sdgSummaries.ShowDialog()
-        sdgSummaries.TestSummaries()
+    Private Sub SetDefaults()
+        ucrBaseDescribeOneVar.clsRsyntax.SetBaseRFunction(clsDefaultFunction.Clone())
+        ucrSelectorDescribeOneVar.Reset()
+        ucrReceiverDescribeOneVar.Clear()
+        SetRCode(Me, ucrBaseDescribeOneVar.clsRsyntax.clsBaseFunction, True)
+        bResetSubdialog = True
         TestOKEnabled()
```
Comment (lilyclements) [inline instat/dlgDescribeOneVariable.vb]: The TestOK doesn't need to be in the SetDefaults(), it just has to be in the Reset() now at the end

File: instat/dlgDescribeOneVariable.vb [hunk 5]
```diff
@@ -29,120 +32,97 @@ Public Class dlgDescribeOneVariable
...
+        'ucrChkSaveResult.SetValuesCheckedAndUnchecked("TRUE", "FALSE")
+        'ucrChkSaveResult.SetRDefault("FALSE")
+
+        'setting controls to appropriate functions
+        ucrChkOmitMissing.SetRCode(clsDefaultFunction, True)
+        ucrChkOmitMissing.SetRCode(clsRCustomFunction, True)
+        ucrChkSaveResult.SetRCode(clsRCustomFunction, True)
+        ucrReceiverDescribeOneVar.SetRCode(clsRCustomFunction, True)
+        ucrReceiverDescribeOneVar.SetRCode(clsDefaultFunction, True)
+        ucrSelectorDescribeOneVar.SetRCode(clsRCustomFunction, True)
```
Comment (lilyclements) [inline instat/dlgDescribeOneVariable.vb]: So, I'm used to having the SetRCode in the reset part. How come it is in here in this case? :)

File: instat/dlgDescribeOneVariable.vb [hunk 6]
```diff
@@ -29,120 +32,97 @@ Public Class dlgDescribeOneVariable
...
+
+        clsDefaultFunction.SetRCommand("summary")
+        clsRCustomFunction.AddParameter("return_output", "TRUE")
+    End Sub
+
     Public Sub TestOKEnabled()
-        If ucrReceiverDescribeOneVar.IsEmpty() OrElse (chkCustomise.Checked AndAlso sdgSummaries.strSummariesParameter = "c()") Then
+        If ucrReceiverDescribeOneVar.IsEmpty() OrElse (ucrChkCustomise.Checked AndAlso sdgSummaries.strSummariesParameter = "c()") Then
```
Comment (lilyclements) [inline instat/dlgDescribeOneVariable.vb]: I know it's disabled at the moment, but perhaps add in here the ucrSave.IsComplete()?

File: instat/dlgDescribeOneVariable.vb [hunk 7]
```diff
@@ -29,120 +32,97 @@ Public Class dlgDescribeOneVariable
...
-        ucrBaseDescribeOneVar.clsRsyntax.iCallType = 2
-        ucrReceiverDescribeOneVar.Selector = ucrSelectorDescribeOneVar
-        ucrReceiverDescribeOneVar.SetMeAsReceiver()
-        clsRCustomFunction.SetRCommand(frmMain.clsRLink.strInstatDataObject & "$summary")
-        clsRBaseFunction.SetRCommand("summary")
-        ucrBaseDescribeOneVar.iHelpTopicID = 410
-    End Sub
-
     Private Sub ucrBaseDescribeOneVar_ClickReset(sender As Object, e As EventArgs) Handles ucrBaseDescribeOneVar.ClickReset
         SetDefaults()
```
Comment (lilyclements) [inline instat/dlgDescribeOneVariable.vb]: In here, there should ordinarily be the SetDefaults(), followed by .SetRCode and then TestOKEnabled(True).
How come the SetRCodeForControls is in the Initialise and not in here out of interest?

**Comment by stevekogo** [comment]
@lilyclements I stopped working on this dialog since some background work on the user controls to be called with different parameter names needs to be done. @dannyparsons can use this as an example to get that functionality working.

**Comment by dannyparsons** [comment]
I've worked on this here #2693 now

**Pull request merged**